backend/cfg/cfgize.ml

```diff
@@ -628,7 +633,7 @@ let rec add_blocks :
       terminate_block ~trap_actions:[]
         (copy_instruction_no_reg state last ~desc:(Cfg.Always label_body));
       let next, add_next_block = prepare_next_block () in
```

Comment:
@xclerc  something strange is going on here. 

- `add_iend_with_poptrap` is recorded for both delayed and regular `Itrywith`
- it is always converted to `Pop_generic` (this is fine because we don't yet use the argument of `Cmm.Pop`)
- `prepare_next_block` modifies state, so moving `add_iend_with_poptrap` earlier (to distinguish the two kinds) breaks the build.
- is it possible to mistakenly generate two `Poptrap` instructions with `flambda2`? one from this `iend_with_pop` and the other from `trap_actions` that come from `Iexit` and `Ireturn`.  I'm thinking some